manager/controllers/package_detail.go

```diff
@@ -0,0 +1,139 @@
+package controllers
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	"net/http"
+)
+
+func singlePackageLatestQuery(pkgName string) *gorm.DB {
+	return database.Db.Table("package pkg").Select("*").Joins("JOIN package_name pn ON pkg.name_id = pn.id").
+		Joins("JOIN advisory_metadata adv ON pkg.advisory_id = adv.id").
+		Where("pn.name = ?", pkgName).
+		Order("adv.public_date DESC").Limit(1)
+}
+
+func singlePackageEvraQuery(pkgName string, evra string) *gorm.DB {
+	return database.Db.Table("package").
+		Joins("JOIN package_name ON package.name_id = package_name.id").
+		Where("package_name.name = ?", pkgName).
+		Where("package.evra = ?", evra)
+}
+
+func getAdvisoryByID(advID int) (models.AdvisoryMetadata, error) {
+	var advisory models.AdvisoryMetadata
+	err := database.Db.Where("id = ?", advID).First(&advisory).Error
+	return advisory, err
+}
+
+type PackageDetailResponse struct {
+	Data PackageDetailItem `json:"data"`
+}
+
+type PackageDetailItem struct {
+	Attributes PackageDetailAttributes `json:"attributes"`
+	ID         string                  `json:"id"`
+	Type       string                  `json:"type"`
+}
+
+type PackageDetailAttributes struct {
+	Description string `json:"description"`
+	Summary     string `json:"summary"`
+	Name        string `json:"name"`
+	Version     string `json:"version"`
+	AdvisoryID  string `json:"advisory_id"`
+}
+
+func packageLatestHandler(c *gin.Context, packageName string) {
+	query := singlePackageLatestQuery(packageName)
```

Comment:
To be correct, firstly we should check whether input is valid package_name, using some regex. If not we should return Bad request instead of directly trying to search for any input string.

Reply:
This should be done in `PackageDetailHandler` function, which should determine what function should be called, so handlers should not care about input.